src/NameRegistry.sol

```diff
@@ -884,28 +908,41 @@ contract NameRegistry is
     //////////////////////////////////////////////////////////////*/
 
     /**
-     * @notice Move the fname from the current owner to the pool and renew it for another year.
+     * @notice Move the fnames froms the current owners to desired destinations and renew them for another week.
      *         Does not work when paused because it calls _transfer.
      *
-     * @param tokenId the uint256 representation of the fname.
+     * @param reclaimActions an array of ReclaimAction structs representing the fnames and their corresponding
+     *           destination addresses.
      */
-    function reclaim(uint256 tokenId) external payable {
+    function reclaim(ReclaimAction[] calldata reclaimActions) external payable {
         // call msg.sender instead of _msgSender() since we don't need meta-tx for admin actions
         // and it reduces our attack surface area
         if (!hasRole(MODERATOR_ROLE, msg.sender)) revert NotModerator();
+        uint256 reclaimActionsLength = reclaimActions.length;
 
-        uint256 _expiry = expiryOf[tokenId];
+        for (uint256 i = 0; i < reclaimActionsLength;) {
+            uint256 tokenId = reclaimActions[i].tokenId;
 
-        // If an fname hasn't been minted, it should be minted instead of reclaimed
-        if (_expiry == 0) revert Registrable();
+            address destination = reclaimActions[i].destination;
```

Comment:
we can avoid this allocation to destination and directly pass it into _transfer on line 940. if that saves gas, it might be worth doing.